#!/usr/bin/env python
PACKAGE = "colorado"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Steer_control_P",	double_t,	0,	"Porptional",	50000,	0,  50000)
gen.add("Steer_control_I",	double_t,	0,	"Integral",	100,    0,  1000)
gen.add("Steer_control_D",	double_t,	0,	"Differential",	800,	0,  10000)	
#Engine power, Velocity damping and steering speed:
gen.add("Power",	double_t,	0,	"Power multiplier",	2,	0,  10000)	
gen.add("Damping",	double_t,	0,	"Velocity Damping",	12,	0,  1000)
gen.add("Friction",	double_t,	0,	"Wheel Dynamic Friction",	100,	0,  1000)	
gen.add("Steering",	double_t,	0,	"Streeing speed",	2,	0,  1000)	
#Suspension system values:
gen.add("Spring",	double_t,	0,	"Suspension Spring",	28000,	0,  100000)	
gen.add("Damper",	double_t,	0,	"Suspesion Damping",	4000,	-5000,  5000)	
gen.add("Target",	double_t,	0,	"Suspesion Target",	0.11,	-2,  2)	

exit(gen.generate(PACKAGE, "colorado_reconfig_node", "colorado"))